### ISSUE TITLE
Deleting a SpriteFrames Animation does not update owning AnimatedSprite's Animation property to an existing Animation

### ISSUE THREAD

**Issue by mymikal** [godotengine/godot#28658]
**Godot version:**
v3.1.1.stable.official

**OS/device including version:**
Windows 10 Home
Version: 1803
Build: 17134.706

**Issue description:**
When I delete a SpriteFrame Animation, the Inspector shows that the Animation is still selected and testing through gdscript reveals that the owning AnimatedSprite's Animation property continues to hold the string name of the deleted Animation until it is manually changed. When changed, the name of the deleted Animation vanishes.

Expected: when I delete a SpriteFrame Animation, the program should update the Inspector to an existing Animation and set the owning AnimatedSprite's Animation property to that Animation's name.

**Steps to reproduce:**
1. Open Godot (see version at top)
2. Create a new project.
3. Create an AnimatedSprite node.
4. Select the AnimatedSprite node.
5. In the Inspector tab, go to Frames and create a new SpriteFrames asset.
6. If the SpriteFrames asset is not already selected, select the SpriteFrames asset in the Inspector.
7. If the Animation window is not already open, select the SpriteFrames option on the bottom of the Godot viewport.
8. Create 2 new animations. Note that when you create a new animation, the inspector will update to select that animation by name.
9. Delete the currently assigned animation. Note that when the currently assigned animation is deleted, *the inspector does not update the Animation property to an existing animation*.

**Additional Steps:**
10. Attach a new empty script to the AnimatedSprite node and insert the following code.
```gdscript
var animation_names = PoolStringArray() # holds string names of frame sequences
var animation_dictionary = Dictionary() # maps animation names to booleans

func _ready():
	animation_names = self.frames.get_animation_names() # get the current animation names
	for animation_name in animation_names: # for each name
		animation_dictionary[animation_name] = false # map false to it
	test()

func test(): # testing block to execute error method
	if is_locked():
		print("locked")
	else:
		print("unlocked")

func is_locked(): # method that will cause error
	return animation_dictionary[self.animation] # get current boolean for a given animation name
```
11. Save the script.
12. Run the scene.

**Result:** the debugger will inform you that you have provided an invalid index to the dictionary, whose string name matches the name of the Animation you deleted.

**Cause:** when you create a new Animation, it is automatically switched to in the Inspector for the AnimationSprite node; however, if you delete that same Animation, the AnimatedSprite node continues to hold the string name of the deleted Animation until you manually select a new one in the Inspector.

**Minimal reproduction project:**
[AnimatedSpriteTest.zip](https://github.com/godotengine/godot/files/3143250/AnimatedSpriteTest.zip)

**Comment by KoBeWi** [comment]
Still valid in 3.2.3 rc3

**Comment by akien-mga** [comment]
Fixed by #49495.
